src/coreclr/src/jit/hwintrinsiccodegenarm64.cpp

```diff
@@ -644,6 +644,27 @@ void CodeGen::genHWIntrinsic(GenTreeHWIntrinsic* node)
                 }
                 break;
 
+            case NI_AdvSimd_InsertScalar:
+            {
+                assert(isRMW);
+                assert(targetReg != op3Reg);
+
+                if (targetReg != op1Reg)
+                {
+                    GetEmitter()->emitIns_R_R(INS_mov, emitTypeSize(node), targetReg, op1Reg);
+                }
+
+                HWIntrinsicImmOpHelper helper(this, intrin.op2, node);
+
+                for (helper.EmitBegin(); !helper.Done(); helper.EmitCaseEnd())
+                {
+                    const int elementIndex = helper.ImmValue();
+
+                    GetEmitter()->emitIns_R_R_I_I(ins, emitSize, targetReg, op3Reg, elementIndex, 0, opt);
+                }
```

Comment:
Can you point me to the test that would exercise this code path?

Reply:
InsertScalar tests I added in this PR

Reply:
Sure, I meant to ask which portion of `InsertScalar` test. I am just curious.

Reply:
The for loop is triggered in both cases - non-const imm and const imm. The difference is how many iterations is done and whether we insert any "header" the instruction(s) emitted for the intrinsic.

Say, for const case it's one iteration and there is no header. While for non const case we emit code that computes branching target given the non const value and emit multiple instructions (for each immValue in [immLowerBound, immUpperBound]) follower by `j endLabel`

Reply:
Ok, thanks!